scripts/benchmark.cr

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env -S crystal run --no-debug --release
+
+require "../src/pluto"
+
+record BenchmarkResult, name : String, time : Int64, memory : Int64
+
+def benchmark_memory
+  bytes_before_measure = GC.stats.total_bytes
+  yield
+  (GC.stats.total_bytes - bytes_before_measure).to_i64
+end
+
+def benchmark_time
+  time = Time.monotonic
+  yield
+  Time.monotonic - time
+end
+
+def print_result_table(benchmarks : Array(BenchmarkResult))
+  name_rjust = benchmarks.map(&.name.size).max
+  time_ljust = benchmarks.map { |result| result.time.to_s.size + 2 }.max
+  memo_ljust = benchmarks.map(&.memory.humanize_bytes.size).max
+
+  # Headers
+  table = [
+    [" " * name_rjust, "Time".ljust(time_ljust), "Memory".ljust(memo_ljust)],
+    ["-" * name_rjust, "-" * time_ljust, "-" * memo_ljust],
+  ]
+
+  # Rows
+  benchmarks.each do |result|
+    table << [
+      result.name.rjust(name_rjust),
+      "#{result.time}ms".ljust(time_ljust),
+      result.memory.humanize_bytes.ljust(memo_ljust),
+    ]
+  end
+
+  output = String.build do |string|
+    table.each do |row|
+      string << "| " << row.join(" | ") << " |\n"
```

Comment:
Note: the leading and trailing `|` characters are required for the table to print nicely in github. Table looks fine in terminal either way, but it's nice to copy / paste it as it is into GH and have it render.